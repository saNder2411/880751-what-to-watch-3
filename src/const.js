const CardMode = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

const DEFAULT_GENRE = `All genres`;

const ShowingCardsAmount = {
  ON_START: 8,
  BY_BUTTON: 8,
};

const DataTypes = {
  PROMO_DATA: `PROMO_DATA`,
  CARDS_DATA: `CARDS_DATA`,
  REVIEWS_DATA: `REVIEWS_DATA`,
};

const ComponentTypes = {
  PREVIEW_CARDS_LIST: `PREVIEW_CARDS_LIST`,
  GENRES_LIST: `GENRES_LIST`,
};

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export {CardMode, DEFAULT_GENRE, ShowingCardsAmount, DataTypes, ComponentTypes, AuthorizationStatus};
