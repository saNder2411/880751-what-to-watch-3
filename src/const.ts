const DEFAULT_GENRE: string = `All genres`;

export {DEFAULT_GENRE};
