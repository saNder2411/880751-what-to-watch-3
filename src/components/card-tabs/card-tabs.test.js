import React from 'react';
import renderer from 'react-test-renderer';
import CardTabs from './card-tabs';
import CardOverview from '../../components/card-overview/card-overview';
import CardDetails from '../../components/card-details/card-details';
import CardReviews from '../../components/card-reviews/card-reviews';
import withCardTabsState from '../../hocs/with-card-tabs-state/with-card-tabs-state';
import withFetchData from '../../hocs/with-fetch-data/with-fetch-data';
import {DataTypes} from '../../const';

const selectedCard = {
  id: 1,
  backgroundImage: `bg-the-grand-budapest-hotel`,
  posterImage: `the-grand-budapest-hotel-poster`,
  previewImage: `img/bohemian-rhapsody.jpg`,
  title: `Bohemian Rhapsody`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  rating: 9,
  scoresCount: 100,
  previewVideoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  director: `Steven Spielberg`,
  starring: [
    `Judi Dench`, `Robert De Niro`, `Leonardo DiCaprio`, `Morgan Freeman`, `Tom Hanks`,
  ],
  runtime: 98,
  genre: `Drama`,
  released: 1989,
};

const WrappedCardTabs = withCardTabsState(CardTabs);
const WrappedCardReviews = withFetchData(DataTypes.FETCH_REVIEWS_DATA)(CardReviews);

it(`Should CardOverview render correctly`, () => {
  const markup = renderer
    .create(
        <WrappedCardTabs >
          <CardOverview {...selectedCard} />
          <CardDetails {...selectedCard} />
          <WrappedCardReviews />
        </WrappedCardTabs>
    )
    .toJSON();

  expect(markup).toMatchSnapshot();
});
