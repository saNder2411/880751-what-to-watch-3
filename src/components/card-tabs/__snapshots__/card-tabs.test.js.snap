// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CardOverview render correctly 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
